### PR TITLE
Release Cadence added

### PR DESCRIPTION
**Pull request by sumawa**

### CODE REVIEW

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 439800.43**

File: RELEASE_CADENCE.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+* ***MAJOR***: Major releases are not on any particular cadence. Major releases are likely to be stable over a long period (one year or more)
+* **FEATURE**: zHttp are targeting new feature releases every fortnightly and will include new features, improvements and bug fixes
+* **PATCH**: This is reserved for urgent bug fixes on the current release
+* **RC(Release Candidate)**: This is reserved for beta version
+* **M(Milestone)**: This is reserved for major Milestone before every prerelease
+
+# Backward compatibility
+We will not support backward compatibility till major stable release
+
+# Change Log templates (WIP - example will be added)
```
Comment (amitksingh1490) [inline RELEASE_CADENCE.md]: Please add the example and remove WIP

File: RELEASE_CADENCE.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+| 0.1.1 (Major bug was found in 0.1)| Fri Dec 24, 2021|
+
+* The above table is a rough example. In reality the dates may change due to holidays, availability of resources etc.
+* The patch fixes (0.1.1) are very important and the triage may take more than 2-3 days depending on the complexity of the bug.
+* Major versions (with possible breaking changes) can be released in 6 months or a year.
+* The creation of a release candidate can be done earlier (example Monday Dec 6)  based on the team discussion
+
+# Versioning
+* zHttp will follow semantic versioning
+* Every pre-release will be versioned: <MAJOR>.<FEATURE>.<PATCH>-<RC | M>
```
Comment (amitksingh1490) [inline RELEASE_CADENCE.md:45]: Doesn't go with our current versioning

File: RELEASE_CADENCE.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ Example:  v1.1.0-RC1   (if previous release was v1.0.x) 
+```
+* All our PRs meant for this release (including mainly stability and bug fixes) are based against this branch. (Remember no new features are included in this branch)
+* The captain will use this vX.Y.Z-rc branch to create a small PR to the main branch and update the CHANGELOG.md on the release day and update version where required.
+* Make sure documentation / examples are updated by authors.
+
+### A Release Example
+| **Release**    | **Date**         |
+| ----------- | ----------- |
+| 0.1.0-rc      | Wed Dec 8, 2021       |
```
Comment (amitksingh1490) [inline RELEASE_CADENCE.md]:
```suggestion
| 1.0.0.0-RC1      | Wed Dec 8, 2021       |
```

File: RELEASE_CADENCE.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+# ZIO-HTTP(zHttp) Release Cadence
+
+- [Pre-Release cycles](#pre-release-cycles)
+    - [Release Highlights](#release-highlights)
+    - [Release Process](#release-process)
+    - [A Release Example](#a-release-example)
+
+- [Versioning](#versioning)
+- [Change log Templates](#change-log-templates-wip---example-will-be-added)
```
Comment (amitksingh1490) [inline RELEASE_CADENCE.md]: Remove WIP add example

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 443474.04**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 436338.84**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 425553.16**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 435986.75**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 434793.77**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 442771.78**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 435635.02**

File: RELEASE_CADENCE.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+# Pre-Release cycles
+
+### Release Highlights
+* zHttp will follow a fortnightly release before first major release
```
Comment (tusharmath) [inline RELEASE_CADENCE.md]:
```suggestion
* ZIO Http will follow a fortnightly release before first major release
```

**Comment by tusharmath** [comment]
- Run through spell check and Grammarly once.

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 434049.41**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 444486.63**

File: RELEASE_CADENCE.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+### Release Highlights
+* ZIO Http will follow a fortnightly release before first major release
+* We aim for regular and strict one release per 2 weeks.
+* 2 weeks are counted from the first release candidate to another.
+* This means that there is no code freeze and no feature should block a release
+
+### Release Process
+* We will target releases every second Wednesday. This will give us time to find some major bugs and do a patch release if required in the next 2-3 days
+* From release/next we will cut a branch following this pattern
```
Comment (amitksingh1490) [inline RELEASE_CADENCE.md]: This line is not needed we will not have any release/next. All PR's will be opened against main

File: RELEASE_CADENCE.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+``` 
+v<major>.<minor+1>.0-RC<+1>
+Example:  v1.1.0-RC1   (if previous release was v1.0.x)
+```
+* All our PRs meant for this release (including mainly stability and bug fixes) are based on this branch. (Remember no new features are included in this branch)
+* The captain will use this vX.Y.Z-RC branch to create a small PR to the main branch and update the CHANGELOG.md on the release day and update the version where required.
```
Comment (amitksingh1490) [inline RELEASE_CADENCE.md]: Captain??? I think this should not be part of cadence document

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 423514.25**

File: RELEASE_CADENCE.md (hunk 5 above)
Comment (sumawa) [reply RELEASE_CADENCE.md]: Done

File: RELEASE_CADENCE.md (hunk 2 above)
Comment (sumawa) [reply RELEASE_CADENCE.md:45]: We will clarify the third 0 in 1.0.0.0-RC1

File: RELEASE_CADENCE.md (hunk 3 above)
Comment (sumawa) [reply RELEASE_CADENCE.md]: Done

File: RELEASE_CADENCE.md (hunk 4 above)
Comment (sumawa) [reply RELEASE_CADENCE.md]: Example added

File: RELEASE_CADENCE.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  - [Guiding Principle](#guiding-principle)
+  - [Example of a Changelog](#example-of-a-changelog)
+
+# Pre-Release cycles
+
+### Release Highlights
+* ZIO Http will follow a fortnightly release before first major release
+* We aim for regular and strict one release per 2 weeks.
+* 2 weeks are counted from the first release candidate to another.
+* This means that there is no code freeze and no feature should block a release
```
Comment (tusharmath) [inline RELEASE_CADENCE.md:19]: It's possible that something is broken in `main` and we are unable to make a release. In that case, should we still continue with the release?

**Comment by sumawa** [comment]
Keeping an internal document for this.

**Pull request closed without merging**